### PR TITLE
Fix depreacated sf3.0

### PR DESCRIPTION
**Pull request by Barbarians**

### CODE REVIEW

File: Notifier/SmsOcitoNotifier.php [hunk 1]
```diff
@@ -158,32 +159,32 @@ public function getContentFields()
     public function getFromFields()
     {
         return array(
-            'userName' => array('text',    array('required' => false, 'max_length' => 30)),
-            'password' => array('text',    array('required' => false, 'max_length' => 30)),
-            'senderType' => array('choice', array(
+            'userName' => array(Types\TextType::class,    array('required' => false, 'max_length' => 30)),
+            'password' => array(Types\TextType::class,    array('required' => false, 'max_length' => 30)),
```
Comment (konandrum) [inline Notifier/SmsOcitoNotifier.php]: remove white spaces

File: Notifier/PushIOSNotifier.php [hunk 2]
```diff
@@ -10,6 +10,8 @@
 
 use IDCI\Bundle\NotificationBundle\Entity\Notification;
 use IDCI\Bundle\NotificationBundle\Exception\PushIOSNotifierException;
+use Symfony\Component\Form\Extension\Core\Type as Types;
+use IDCI\Bundle\NotificationBundle\Form\Type\CertificateType;
```
Comment (konandrum) [inline Notifier/PushIOSNotifier.php]: alpha order

File: Form/Type/NotifierChoiceType.php [hunk 3]
```diff
@@ -9,6 +9,8 @@
 
 use Symfony\Component\Form\AbstractType;
 use Symfony\Component\OptionsResolver\OptionsResolverInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+use Symfony\Component\Form\Extension\Core\Type\ChoiceType;
```
Comment (konandrum) [inline Form/Type/NotifierChoiceType.php]: alpha order

File: Form/Type/CertificateType.php [hunk 4]
```diff
@@ -11,6 +11,7 @@
 use Symfony\Component\Form\FormBuilderInterface;
 use Symfony\Component\Form\FormEvent;
 use Symfony\Component\Form\FormEvents;
+use Symfony\Component\Form\Extension\Core\Type as Types;
```
Comment (konandrum) [inline Form/Type/CertificateType.php]: alpha order

File: Form/NotifierTypeSelectionType.php [hunk 5]
```diff
@@ -10,6 +10,8 @@
 use Symfony\Component\Form\AbstractType;
 use Symfony\Component\Form\FormBuilderInterface;
 use Symfony\Component\OptionsResolver\OptionsResolverInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+use IDCI\Bundle\NotificationBundle\Form\Type\NotifierChoiceType;
```
Comment (konandrum) [inline Form/NotifierTypeSelectionType.php]: alpha order

File: Form/NotifierConfigurationType.php [hunk 6]
```diff
@@ -10,6 +10,13 @@
 use Symfony\Component\Form\AbstractType;
 use Symfony\Component\Form\FormBuilderInterface;
 use Symfony\Component\OptionsResolver\OptionsResolverInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+use Symfony\Component\Form\Extension\Core\Type as Types;
+use IDCI\Bundle\SimpleMetadataBundle\Form\Type\RelatedToManyMetadataType;
+use Symfony\Component\Form\FormEvent;
+use Symfony\Component\Form\FormEvents;
+use IDCI\Bundle\SimpleMetadataBundle\Form\Type\MetadataType;
+use Symfony\Component\Form\Extension\Core\Type\HiddenType;
```
Comment (konandrum) [inline Form/NotifierConfigurationType.php]: alpha order

File: Form/NotificationType.php [hunk 7]
```diff
@@ -10,7 +10,12 @@
 use Symfony\Component\Form\AbstractType;
 use Symfony\Component\Form\FormBuilderInterface;
 use Symfony\Component\OptionsResolver\OptionsResolverInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
 use IDCI\Bundle\NotificationBundle\Entity\Notification;
+use Symfony\Component\Form\Extension\Core\Type as Types;
+use Symfony\Component\Form\FormEvent;
+use Symfony\Component\Form\FormEvents;
+use IDCI\Bundle\SimpleMetadataBundle\Form\Type\MetadataType;
```
Comment (konandrum) [inline Form/NotificationType.php]: alpha order

File: DependencyInjection/Compiler/NotifierCompilerPass.php [hunk 8]
```diff
@@ -11,7 +11,6 @@
 use Symfony\Component\DependencyInjection\ContainerBuilder;
 use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
 use Symfony\Component\DependencyInjection\Reference;
-use Symfony\Component\DependencyInjection\DefinitionDecorator;
 use IDCI\Bundle\NotificationBundle\Exception\UndefindedDefinitionException;
```
Comment (konandrum) [inline DependencyInjection/Compiler/NotifierCompilerPass.php]: alpha order

File: Notifier/SmsOcitoNotifier.php [hunk 9]
```diff
@@ -158,32 +159,32 @@ public function getContentFields()
...
                     '/SendWapMTRequest.jsp' => 'Send wap mt request',
                     '/SendMailMTRequest.jsp' => 'Send mail mt request',
                 ),
             )),
-            'senderAppId' => array('text',    array('required' => false, 'max_length' => 10)),
-            'senderId' => array('text',    array('required' => false, 'max_length' => 11)),
-            'flag' => array('integer', array(
+            'senderAppId' => array(Types\TextType::class,    array('required' => false, 'max_length' => 10)),
+            'senderId' => array(Types\TextType::class,    array('required' => false, 'max_length' => 11)),
+            'flag' => array(Types\IntegerType::class, array(
```
Comment (konandrum) [inline Notifier/SmsOcitoNotifier.php]: remove white spaces

File: Resources/config/services.yml [hunk 10]
```diff
@@ -37,17 +37,6 @@ services:
...
-        class: IDCI\Bundle\NotificationBundle\Form\Type\AbstractNotificationType
-        arguments: [null, null]
-        abstract:  true
-
-    idci_notification.form.type.abstract_notifier_configuration:
-        class: IDCI\Bundle\NotificationBundle\Form\Type\AbstractNotifierConfigurationType
-        arguments: [null, null]
-        abstract:  true
-
-
```
Comment (konandrum) [inline Resources/config/services.yml:50]: Why ? (cf compilerpass)

File: DependencyInjection/Compiler/NotifierCompilerPass.php [hunk 11]
```diff
@@ -58,18 +57,10 @@ public function process(ContainerBuilder $container)
                 // Define Notification form
                 $notificationFormAlias = sprintf('notification_%s', $alias);
                 $notificationFormServiceId = sprintf('idci_notification.form.type.%s', $notificationFormAlias);
-                $notificationFormDefinition = new DefinitionDecorator('idci_notification.form.type.abstract_notification');
-                $notificationFormDefinition->replaceArgument(0, $alias);
-                $notificationFormDefinition->replaceArgument(1, $notifierReference);
-                $container->setDefinition($notificationFormServiceId, $notificationFormDefinition);
```
Comment (konandrum) [inline DependencyInjection/Compiler/NotifierCompilerPass.php:64]: Why

File: DependencyInjection/Compiler/NotifierCompilerPass.php [hunk 12]
```diff
@@ -58,18 +57,10 @@ public function process(ContainerBuilder $container)
...
 
                 // Define NotifierConfiguration form
                 $notifierConfigurationFormAlias = sprintf('notifier_configuration_%s', $alias);
                 $notifierConfigurationFormServiceId = sprintf('idci_notification.form.type.%s', $notifierConfigurationFormAlias);
-                $notifierConfigurationFormDefinition = new DefinitionDecorator('idci_notification.form.type.abstract_notifier_configuration');
-                $notifierConfigurationFormDefinition->replaceArgument(0, $alias);
-                $notifierConfigurationFormDefinition->replaceArgument(1, $notifierReference);
-                $container->setDefinition($notifierConfigurationFormServiceId, $notifierConfigurationFormDefinition);
```
Comment (konandrum) [inline DependencyInjection/Compiler/NotifierCompilerPass.php:72]: Why

**Review by konandrum — CHANGES_REQUESTED** [review]

File: DependencyInjection/Compiler/NotifierCompilerPass.php (hunk 11 above)
Comment (Barbarians) [reply DependencyInjection/Compiler/NotifierCompilerPass.php:64]: i have remove AbstractNotificationType & AbstractNotifierType class and add a event listener on PRE_SET_DATA for manage field with type selected

https://github.com/Tessi-Tms/NotificationBundle/blob/master/Form/NotificationType.php#L42-L75
https://github.com/Tessi-Tms/NotificationBundle/blob/master/Form/NotifierConfigurationType.php#L31-L44

with sf3.0 on createForm i need to call type with "::class" your abstract containt a construct but is not callable.

https://github.com/Tessi-Tms/NotificationManager/blob/master/src/Tms/WebAdminBundle/Controller/NotificationController.php#L99-L101

File: DependencyInjection/Compiler/NotifierCompilerPass.php (hunk 12 above)
Comment (Barbarians) [reply DependencyInjection/Compiler/NotifierCompilerPass.php:72]: i have remove AbstractNotificationType & AbstractNotifierType class and add a event listener on PRE_SET_DATA for manage field with type selected

https://github.com/Tessi-Tms/NotificationBundle/blob/master/Form/NotificationType.php#L42-L75
https://github.com/Tessi-Tms/NotificationBundle/blob/master/Form/NotifierConfigurationType.php#L31-L44

with sf3.0 on createForm i need to call type with "::class" your abstract containt a construct but is not callable.

https://github.com/Tessi-Tms/NotificationManager/blob/master/src/Tms/WebAdminBundle/Controller/NotificationController.php#L99-L101

File: Resources/config/services.yml (hunk 10 above)
Comment (Barbarians) [reply Resources/config/services.yml:50]: i have remove AbstractNotificationType & AbstractNotifierType class and add a event listener on PRE_SET_DATA for manage field with type selected

https://github.com/Tessi-Tms/NotificationBundle/blob/master/Form/NotificationType.php#L42-L75
https://github.com/Tessi-Tms/NotificationBundle/blob/master/Form/NotifierConfigurationType.php#L31-L44

with sf3.0 on createForm i need to call type with "::class" your abstract containt a construct but is not callable.

https://github.com/Tessi-Tms/NotificationManager/blob/master/src/Tms/WebAdminBundle/Controller/NotificationController.php#L99-L101

**Pull request closed without merging**